src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php

```diff
@@ -76,8 +76,13 @@ public function locateSpecifications(Suite $suite, $locator)
 
         if ($locator) {
             $filters = array(new PathsFilter($suiteLocators));
+            $features = array();
+            foreach ($this->findFeatureFiles($locator) as $featureFile) {
+                // Collect all the feature specification files.
+                $features = array_merge($features, $this->gherkin->load($featureFile, $filters));
```

Comment:
i imagine this to be quite slow for big projects. wondering if we can improve it somehow.

Reply:
Not sure. In my case the $locator is an exact feature file path. In this case foreach will loop only 1 time.
In addiition I guess it will be as "slow" as the next foreach block in this file below.